### DISCUSSION TITLE
Song progress text doesn't reflect rate changes

### DISCUSSION THREAD

**Discussion by Cwazywierdo** [ppy/osu#22731] [Q&A] [answered]
Currently the song progress text shows the current/remaining times unadjusted for rate changing mods, which in the simple case of HT/DT I think should be changed. For wind up/down, the times can technically be calculated but it'd be much easier just to group them with adaptive speed under a `INonConstantRateChange` interface. Or maybe the other way around with HT/DT getting `IConstantRateChange`.

**Comment by Theighlin** [comment, marked as answer]
Duplicate of #8929
